Zend/zend_language_scanner.l

```diff
@@ -1250,6 +1258,64 @@ static void copy_heredoc_label_stack(void *void_heredoc_label)
 	zend_ptr_stack_push(&SCNG(heredoc_label_stack), (void *) new_heredoc_label);
 }
 
+/* Check that { }, [ ], ( ) are nested correctly */
+static void report_bad_nesting(char opening, int opening_lineno, char closing)
+{
+	char   buf[256];
+	size_t used = 0;
+
+	used = snprintf(buf, sizeof(buf), "Unclosed '%c'", opening);
+
+	if (opening_lineno != CG(zend_lineno)) {
+		used += snprintf(buf + used, sizeof(buf) - used, " on line %d", opening_lineno);
+	}
+
+	if (closing) { 	/* 'closing' will be 0 if at end of file */
+		used += snprintf(buf + used, sizeof(buf) - used, " does not match '%c'", closing);
+	}
+
+	zend_throw_exception(zend_ce_parse_error, buf, 0);
+}
+
+static void enter_nesting(char opening)
+{
+	zend_nest_location nest_loc = {opening, CG(zend_lineno)};
+	zend_stack_push(&SCNG(nest_location_stack), &nest_loc);
+}
+
+static int exit_nesting(char closing)
+{
+	if (zend_stack_is_empty(&SCNG(nest_location_stack))) {
+		zend_throw_exception_ex(zend_ce_parse_error, 0, "Unmatched '%c'", closing);
+		return -1;
+	}
+
+	zend_nest_location *nest_loc = zend_stack_top(&SCNG(nest_location_stack));
+	char opening = nest_loc->text;
+	char closing = *location;
```

Comment:
This leftover is currently making the build fail.

Reply:
Thanks. I actually saw that already and am just running `make` and `make test`.

Reply:
Sorry for my haste to push out the last update!

Reply:
By the way, there is some funniness with PHP's Makefiles. I find that after editing certain source files, it is necessary to `make clean`. Just `make` doesn't rebuild everything which needs to be rebuilt, and then some tests fail with SIGTERM, SIGSEGV, etc.

Reply:
The makefiles don't track header dependencies, so this can happen when data structures in headers are changed.

It would be nice to implement this, but everyone is afraid to touch the build system :)